JuliaStream.jl/src/StreamData.jl

```diff
@@ -0,0 +1,8 @@
+
```

Comment:
Why do we need to wrap up the data arrays in a whole new struct?

Reply:
It's similar to what we have for Rust, there are no classes in Julia, only functions and data (structs).
The alternative is to pass everything as parameters:
```
function mul!(data::StreamData{T}) where {T}
```
becomes
```
function mul!(a::C, b::C, c::C, scalar::T, size::Int) where {T,C<:AbstractArray{T}}
```
Same applies when I copy everything back to host:
```
const ROCData = StreamData{T,ROCArray{T}} where {T}
...
function read_data(data::ROCData{T})::VectorData{T} where {T}
  return VectorData{T}(data.a, data.b, data.c, data.scalar, data.size)
end
```
`read_data` would be very long, and it needs to be implemented for 5 different implementations.

Reply:
I think I prefer passing things as parameters.